pkg/authorization/registry/subjectaccessreview/rest.go

```diff
@@ -112,19 +118,33 @@ func (r *REST) Create(ctx kapi.Context, obj runtime.Object) (runtime.Object, err
 }
 
 // isAllowed checks to see if the current user has rights to issue a LocalSubjectAccessReview on the namespace they're attempting to access
-func (r *REST) isAllowed(ctx kapi.Context, sar *authorizationapi.SubjectAccessReview) error {
-	localSARAttributes := authorizer.DefaultAuthorizationAttributes{
-		Verb:              "create",
-		Resource:          "localsubjectaccessreviews",
-		RequestAttributes: sar,
+func (r *REST) isAllowed(user user.Info, sar *authorizationapi.SubjectAccessReview) error {
+	var localSARAttributes kauthorizer.AttributesRecord
+	// if they are running a personalSAR, create synthentic check for selfSAR
+	if authorizer.IsPersonalAccessReviewFromSAR(sar) {
+		localSARAttributes = kauthorizer.AttributesRecord{
```

Comment:
This and every other `AttributesRecord` that is manually built is incorrect.  `ResourceRequest` needs to be explicitly set to `true`.

```
I0309 01:03:37.003518   13555 messages.go:126] User "developer" cannot "create" on "" <nil> authorizer.AttributesRecord{User:(*user.DefaultInfo)(0xc4277905c0), Verb:"create", Namespace:"", APIGroup:"authorization.k8s.io", APIVersion:"", Resource:"selfsubjectaccessreviews", Subresource:"", Name:"", ResourceRequest:false, Path:""}
```